### PR TITLE
[MX-504] Added ES version check for IE11 compatibility

### PR DESCRIPTION
**Pull request by karkevin**
# Description
- bandaid fix for ES version compatibility for IE11 users
- added git pre commit check using eslint and husky
- No clean way of checking for ES version in `.phtml` files using composer, so added some npm dev dependencies. These can be removed once refactoring to move JS files out of M2 into source is accomplished. Dependencies can then be accessed from source. 

- A more sustainable change would be to use a transpiler (such as babel and webpack) to convert es6+ syntax into es5 during the build stage. This solution can be further looked into after the JS refactor, and assets are loaded into M2 from source.

Fixes: [MX-504](https://boltpay.atlassian.net/browse/MX-504)

#changelog [MX-504] Added ES version check for IE11 compatibility
- added pre commit hooks and linting instructions
- modified JS from es6 to es5 syntax in `boltaccount.phtml`

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [X] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [X] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: view/frontend/templates/boltaccount.phtml [hunk 1]
```diff
@@ -51,7 +51,7 @@ $checkoutKey = $block->getCheckoutKey();
                 // When magento constructs mobile menu it copies elements instead of moving
                 // We need to wait for this coping, remove first bolt-account-login div
                 // And only ofter that insert account.js script
-                let timerId = setTimeout(function boltAccountLookup() {
+                var timerId = setTimeout(function boltAccountLookup() {
```
Comment (njfaries) [inline view/frontend/templates/boltaccount.phtml:54]: This seems unrelated?

Comment (njfaries) [reply view/frontend/templates/boltaccount.phtml:54]: nvm missed bit in description

Comment (karkevin) [reply view/frontend/templates/boltaccount.phtml:54]: @njfaries `let` is a keyword present in es6, which is not supported in es5 for defining variables. Unsure if it causes any problems with IE, but the change is required in order to fulfill the pre-commit hook.

Comment (karkevin) [reply view/frontend/templates/boltaccount.phtml:54]: This shouldn't be an issue regardless, since `timerId` is only defined in the block and not called globally.
